--- a/sys/dev/pci/pci_dw.c
+++ b/sys/dev/pci/pci_dw.c
@@ -434,7 +434,13 @@ pci_dw_decode_ranges(struct pci_dw_softc *sc, struct ofw_pci_range *ranges,
 	nmem = 0;
 	for (i = 0; i < nranges; i++) {
+		/*
+		 * RK3568 assigns endpoint BARs from its prefetchable MEM64 range.
+		 * It needs an outbound iATU window just like a MEM32 range.
+		 */
 		if ((ranges[i].pci_hi & OFW_PCI_PHYS_HI_SPACEMASK) ==
-		    OFW_PCI_PHYS_HI_SPACE_MEM32)
+		    OFW_PCI_PHYS_HI_SPACE_MEM32 ||
+		    (ranges[i].pci_hi & OFW_PCI_PHYS_HI_SPACEMASK) ==
+		    OFW_PCI_PHYS_HI_SPACE_MEM64)
 			++nmem;
 	}
 
@@ -464,8 +470,10 @@ pci_dw_decode_ranges(struct pci_dw_softc *sc, struct ofw_pci_range *ranges,
 					sc->io_range.size = UINT32_MAX;
 			}
 		}
-		if ((ranges[i].pci_hi & OFW_PCI_PHYS_HI_SPACEMASK) ==
-		    OFW_PCI_PHYS_HI_SPACE_MEM32) {
+		if (((ranges[i].pci_hi & OFW_PCI_PHYS_HI_SPACEMASK) ==
+		    OFW_PCI_PHYS_HI_SPACE_MEM32) ||
+		    ((ranges[i].pci_hi & OFW_PCI_PHYS_HI_SPACEMASK) ==
+		    OFW_PCI_PHYS_HI_SPACE_MEM64)) {
 			MPASS(nmem < sc->num_mem_ranges);
 			sc->mem_ranges[nmem] = ranges[i];
 			if (sc->mem_ranges[nmem].size > UINT32_MAX) {
